dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/datavalue/DataValueAuditServiceTest.java

```diff
@@ -250,4 +250,23 @@ void testGetDataValueAuditNoResult()
 
         assertEquals( 0, dataValueAuditService.getDataValueAudits( params ).size() );
     }
+
+    @Test
+    void testGetDataValueAuditWithFakeCreate()
+    {
+        DataValueAuditQueryParams params = new DataValueAuditQueryParams()
+            .setDataElements( List.of( dataElementA ) )
+            .setPeriods( List.of( periodD ) )
+            .setOrgUnits( List.of( orgUnitA ) )
+            .setCategoryOptionCombo( optionCombo )
+            .setAuditTypes( List.of( AuditType.UPDATE ) );
+
+        assertEquals( 0, dataValueAuditService.getDataValueAudits( params ).size() );
+
+        List<DataValueAudit> audits = dataValueAuditService.getDataValueAudits(
+            dataElementA, periodA, orgUnitA, optionCombo, optionCombo );
+
+        assertEquals( 1, audits.size() );
```

Comment:
There is also `Assertions # assertContainsOnly` now which gives more accurate assertions and better debugging/logging.